docs/reference/image_plugins.rst

```diff
@@ -0,0 +1,59 @@
+============================================
+Registering external image types via plugins
+============================================
+
+
+Usage
+~~~~~
+
+Toga can be extended, via plugins, to understand externally defined image types, gaining
+the ability to convert them to and from its own :any:`toga.Image` class.
+
+A plugin defining how to handle an image format must define three things:
+
+* ``image_class``
+
+  The class representing an image that we want to tell Toga about. For our example, it
+  would be assigned ``MyImage``. (You will, of course, have to import the relevant
+  class in order to refer to it.)
+
+* ``convert_from_format(image_in_format)``
+
+  This top-level function should accept an image of the relevant image class, and return a
+  bytes-like object representing the image in PNG format.
```

Comment:
Is the PNG format strictly a requirement? It's definitely going to be the *safest* option... but other format should work as long as they're widely understood... I'd be surprised if JPG format and GIF didn't work everywhere out of the box.

Reply:
I think I was just thinking of the PIL conversion implementation you provided, which used PNG. I've changed this to a link to the supported image types, does that work?